frappe/integrations/doctype/dropbox_settings/dropbox_settings.py

```diff
@@ -135,7 +135,7 @@ def upload_from_folder(path, is_private, dropbox_folder, dropbox_client, did_not
 
 	for f in frappe.get_all("File", filters={"is_folder": 0, "is_private": is_private,
 		"uploaded_to_dropbox": 0}, fields=['file_url', 'name', 'file_name']):
-		if not f.file_url:
+		if not f.file_url or not f.file_name:
```

Comment:
A better fix would be to verify the file url instead of assuming all files with file name would be correct.
Also file_name is not a mandatory field

Reply:
file_name is not mandatory because it is set explicitly in the next line `filename = f.file_url.rsplit('/', 1)[-1]`
@hasnain2808 Maybe I can check if `filename` is set or not which will be the same fix as this Ig.

Reply:
The comment about file name was a side comment 
Real concern is that the file url can be incorrect regardless of what file name is
The file url itself should be verified instead of depending on 'file name'.